packages/aws-cdk-lib/aws-sns/lib/subscription.ts

```diff
@@ -152,10 +160,85 @@ export class Subscription extends Resource {
       region: props.region,
       redrivePolicy: this.buildDeadLetterConfig(this.deadLetterQueue),
       subscriptionRoleArn: props.subscriptionRoleArn,
+      deliveryPolicy: props.deliveryPolicy ? this.renderDeliveryPolicy(props.deliveryPolicy, props.protocol): undefined,
     });
 
   }
 
+  private renderDeliveryPolicy(deliveryPolicy: DeliveryPolicy, protocol: SubscriptionProtocol): any {
+    if (protocol !== SubscriptionProtocol.HTTP && protocol !== SubscriptionProtocol.HTTPS) {
+      throw new Error(`Delivery policy is only supported for HTTP and HTTPS subscriptions, got: ${protocol}`);
+    }
+    const { healthyRetryPolicy, throttlePolicy } = deliveryPolicy;
+    if (healthyRetryPolicy) {
+      const delayTargetLimitSecs = 3600;
+      const minDelayTarget = healthyRetryPolicy.minDelayTarget;
+      const maxDelayTarget = healthyRetryPolicy.maxDelayTarget;
+      if (minDelayTarget !== undefined) {
+        if (minDelayTarget.toMilliseconds() % 1000 !== 0) {
+          throw new Error(`minDelayTarget must be a whole number of seconds, got: ${minDelayTarget}`);
+        }
+        const minDelayTargetSecs = minDelayTarget.toSeconds();
+        if (minDelayTargetSecs < 1 || minDelayTargetSecs > delayTargetLimitSecs) {
+          throw new Error(`minDelayTarget must be between 1 and ${delayTargetLimitSecs} seconds inclusive, got: ${minDelayTargetSecs}s`);
+        }
+      }
+      if (maxDelayTarget !== undefined) {
+        if (maxDelayTarget.toMilliseconds() % 1000 !== 0) {
+          throw new Error(`maxDelayTarget must be a whole number of seconds, got: ${maxDelayTarget}`);
+        }
+        const maxDelayTargetSecs = maxDelayTarget.toSeconds();
+        if (maxDelayTargetSecs < 1 || maxDelayTargetSecs > delayTargetLimitSecs) {
+          throw new Error(`maxDelayTarget must be between 1 and ${delayTargetLimitSecs} seconds inclusive, got: ${maxDelayTargetSecs}s`);
+        }
+        if ((minDelayTarget !== undefined) && minDelayTarget.toSeconds() > maxDelayTargetSecs) {
+          throw new Error('minDelayTarget must not exceed maxDelayTarget');
+        }
+      }
+
+      const numRetriesLimit = 100;
```

Comment:
It's good intention to not use the magic value and define it explicitly. But given the number of validations, I actually think it's more readable to directly use the value like 100 and 3600 in the if condition checks. What do you think?

Reply:
I think I prefer it how it is now, I agree it might be more readable, but I'd prefer keeping hard-coding to a minimum to make refactors safer/simplier, etc